src/Microsoft.Diagnostics.Runtime/Native/NativeTypes.cs

```diff
@@ -26,9 +26,12 @@ public NativeType(NativeHeap heap, int index, NativeModule module, string name,
             _eeType = eeType;
             _index = index;
 
```

Comment:
Why are these changes necessary?  ConstructObjectType checks that mtData is not null before calling here.

The only place that could generate a null mtData for this function is CreateFreeType:

            IMethodTableData mtData = NativeRuntime.GetMethodTableData(free);
            _free = new NativeType(this, _types.Count, _mrtModule, "Free", free, mtData);

However, if getting method table data from the free EEType is returning null...then again we have a bad crash dump.  If you are sure we want to handle cases where this fails (I assume for micro crash dumps?), then I think this is ok.